cmd/build.go

```diff
@@ -157,6 +184,87 @@ func build(projectName string, targetOS string, vmArguments []string) {
 	} else {
 		engineCachePath = enginecache.ValidateOrUpdateEngine(targetOS)
 	}
+	dockerFilePath, err := filepath.Abs(filepath.Join(tmpDir, "Dockerfile"))
+	if err != nil {
+		fmt.Printf("hover: Failed to resolve absolute path for Dockerfile %s: %v\n", dockerFilePath, err)
+		os.Exit(1)
+	}
+
+	dockerFile, err := os.Create(dockerFilePath)
+	if err != nil {
+		fmt.Printf("hover: Failed to create Dockerfile %s: %v\n", dockerFilePath, err)
+		os.Exit(1)
+	}
+	dockerFileContent := []string{
+		"FROM dockercore/golang-cross",
+		"RUN apt-get install libgl1-mesa-dev xorg-dev -y",
+		"WORKDIR /app",
+		"CMD " + strings.Join(buildCommand(targetOS, vmArguments, "build/outputs/"+targetOS+"/"+outputBinaryName(projectName, targetOS)), " "),
+	}
+	for _, env := range buildEnv(targetOS, "/engine") {
+		dockerFileContent = append(dockerFileContent, "ENV "+env)
+	}
+
+	for _, line := range dockerFileContent {
+		if _, err := dockerFile.WriteString(line + "\n"); err != nil {
+			fmt.Printf("hover: Could not write Dockerfile: %v\n", err)
+			os.Exit(1)
+		}
+	}
+	err = dockerFile.Close()
+	if err != nil {
+		fmt.Printf("hover: Could not close Dockerfile: %v\n", err)
+		os.Exit(1)
+	}
+	dockerBuildCmd := exec.Command(dockerBin, "build", "-t", "hover-build-cc", ".")
+	dockerBuildCmd.Stderr = os.Stderr
+	dockerBuildCmd.Stdout = os.Stdout
+	dockerBuildCmd.Dir = tmpDir
+	err = dockerBuildCmd.Run()
+	if err != nil {
+		fmt.Printf("hover: Docker build failed: %v\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Println("hover: Cross-Compiling 'go-flutter' and plugins using docker")
+
+	outputPath, err := filepath.Abs(filepath.Join(buildPath, "build", "outputs"))
+	dockerRunCmd := exec.Command(dockerBin, "run", "-e", "USERID=$UID", "-v", goPath+":/go", "-v", pkgRootDir+":/app", "-v", engineCachePath+":/engine", "-v", outputPath+":/app/build/outputs", "-v", filepath.Join(userCacheDir, "go-build")+":/cache", "hover-build-cc")
+	dockerRunCmd.Stderr = os.Stderr
+	dockerRunCmd.Stdout = os.Stdout
+	dockerRunCmd.Dir = tmpDir
+	err = dockerRunCmd.Run()
+	if err != nil {
+		fmt.Printf("hover: Docker run failed: %v\n", err)
+		os.Exit(1)
+	}
+	err = os.RemoveAll(tmpDir)
+	if err != nil {
+		fmt.Printf("hover: Could not remove temporary build folder: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func build(projectName string, targetOS string, vmArguments []string) {
+	mingwGccBin, err := exec.LookPath(mingwGccBinName)
+	mingwGppBin, err := exec.LookPath(mingwGppBinName)
+	clangBin, err := exec.LookPath(clangBinName)
+	crossCompile = targetOS != runtime.GOOS
+	if crossCompile {
+		fmt.Printf("hover: Cross-compiling for %s using docker is very experimental\n", targetOS)
+		if mingwGccBin != "" && mingwGppBin != "" {
+			windowsCrossCompileToolsAvailable = true
+		}
+		if clangBin != "" {
+			darwinCrossCompileToolsAvailable = true
+		}
```

Comment:
`windowsCrossCompileToolsAvailable` and `darwinCrossCompileToolsAvailable` are unused

Reply:
OK my idea doesn't work. So just going to remove it.